Packs/SEKOIAIntelligenceCenter/Integrations/SEKOIAIntelligenceCenter/SEKOIAIntelligenceCenter.py

```diff
@@ -0,0 +1,614 @@
+            method="GET",
+            url_suffix="/v2/inthreat/indicators",
+            params={"value": value, "type": indicator_type},
+        )
+
+    def get_indicator_context(self, value: str, indicator_type: str) -> dict:
+        """Get context around the indicators matching the given value
+
+        :type value: ``str``
+        :param value: indicator to get the context for
+
+        :type type: ``str``
+        :param type: type of the indicator
+
+        :return: dict containing the context as returned from the API
+        :rtype: ``Dict[str, Any]``
+        """
+
+        return self._http_request(
+            method="GET",
+            url_suffix="/v2/inthreat/indicators/context",
+            params={"value": value, "type": indicator_type},
+        )
+
+
+""" HELPER FUNCTIONS """
+
+
+def get_reputation_score(indicator_types: list[str]) -> int:
+    """
+    Return DBotScore based on indicator_types.
+    Default score is Unknown
+    """
+
+    for indicator_type in indicator_types:
+        dbotscore = DBOTSCORE_MAPPING.get(indicator_type)
+        if dbotscore:
+            return dbotscore
+
+    return Common.DBotScore.NONE
+
+
+def extract_file_indicator_hashes(file_name: str) -> dict:
+    """
+    Extract hashes composing the STIX indicator name
+    """
+    hashes = {"sha256": "", "sha1": "", "md5": ""}
+
+    # Remove enclosing brackets
+    file_name = file_name.replace("[", "").replace("]", "")
+    stix_object_hashes = file_name.split("OR")
+    for object_hash in stix_object_hashes:
+        if "SHA-256" in object_hash:
+            hashes["sha256"] = object_hash.split("=")[1].strip(" ").strip("'")
+        if "SHA-1" in object_hash:
+            hashes["sha1"] = object_hash.split("=")[1].strip(" ").strip("'")
+        if "MD5" in object_hash:
+            hashes["md5"] = object_hash.split("=")[1].strip(" ").strip("'")
+
+    return hashes
+
+
+def get_reliability_score(confidence: int) -> str:
+    # Based on "Admiralty Credibility" table see
+    # https://docs.oasis-open.org/cti/stix/v2.1/os/stix-v2.1-os.html#_1v6elyto0uqg
+
+    if confidence in range(80, 100):
+        return DBotScoreReliability.A_PLUS
+    if confidence in range(60, 79):
+        return DBotScoreReliability.A
+    if confidence in range(40, 59):
+        return DBotScoreReliability.B
+    if confidence in range(20, 39):
+        return DBotScoreReliability.C
+    if confidence in range(1, 19):
+        return DBotScoreReliability.D
+    if confidence == 0:
+        return DBotScoreReliability.E
+    else:
+        return DBotScoreReliability.F
+
+
+def get_tlp(object_marking_refs: list[str], stix_bundle: dict) -> str:
+    """
+    Retrieve marking-definition object from object_marking_refs and return TLP
+    If no TLP where found, return "RED" by default
+    """
+    for object_marking_ref in object_marking_refs:
+        for stix_object in stix_bundle["objects"]:
+            if stix_object["id"] == object_marking_ref:
```

Comment:
Ensure the marking-definition is a TLP:
```suggestion
            if stix_object["id"] == object_marking_ref and stix_object["definition_type"] == "tlp:
```

Reply:
done